src/apps/core/app.js

```diff
@@ -89,13 +90,35 @@ define(function(require) {
 			self.displayFavicon();
 			self.loadSVG();
 
+			if (crossSiteMessaging !== undefined) {
+				self.bindCrossSiteMessagingHandler(crossSiteMessaging);
+			}
+
 			container.append(mainTemplate);
 
 			monster.waterfall([
 				_.bind(self.loadIncludes, self)
 			], _.bind(self.loadAuth, self));
 		},
 
+		bindCrossSiteMessagingHandler: function(crossSiteMessaging) {
+			const { origin, topics } = crossSiteMessaging;
+
+			window.addEventListener('message', handler);
+
+			function handler(event) {
+				if (event.origin !== origin || !topics.includes(event.data)) {
+					return;
+				}
+
+				if (!monster.util.isAuthorizedTopicForCrossSiteMessaging(event.data)) {
```

Comment:
The function `isAuthorizedTopicForCrossSiteMessaging` currently does not receive any parameters.

Reply:
My bad totally missed this one.